### PR TITLE
[mini] Handle magic interfaces in common_trampoline as if they were variant.

### PR DESCRIPTION
**Pull request by kumpera**

### CODE REVIEW

**Comment by kumpera** [comment]
I only need @vargaz to review this one.

File: mono/mini/mini-trampolines.c [hunk 1]
```diff
@@ -567,6 +567,11 @@ common_call_trampoline (mgreg_t *regs, guint8 *code, MonoMethod *m, MonoVTable *
 			vtable_slot = mini_resolve_imt_method (vt, vtable_slot, imt_method, &impl_method, &addr, &need_rgctx_tramp, &variant_iface, error);
 			return_val_if_nok (error, NULL);
 
+			/* We must handle magic interfaces on rank 1 arrays of ref types as if they were variant */
+			if (!variant_iface && vt->klass->rank == 1 && !vt->klass->element_class->valuetype && imt_method->klass->is_array_special_interface) {
+				variant_iface = imt_method;
+			}
```
Comment (vargaz) [inline mono/mini/mini-trampolines.c]: This doesn't need braces, otherwise lgtm.

**Comment by luhenry** [comment]
@kumpera once merged, can you please backport to all relevant release branches?

**Comment by kumpera** [comment]
@luhenry sure, which ones?

**Comment by marek-safar** [comment]
@kumpera I did 2017-08 and 2017-06. We also need d15-4-2017-06 and d15-4-2017-04 if you think it's safe to backport

**Comment by kumpera** [comment]
@marek-safar It's a simple and safe change. 

d15-4-2017-04 https://github.com/mono/mono/pull/5507
d15-4-2017-06 https://github.com/mono/mono/pull/5506

**Pull request merged**